### PR TITLE
Bugs

### PR DESCRIPTION
**Pull request by matthewl20**
Added clean types to the Relationship model
Validates Base and Related types are in CLEAN_TYPES
Added relationship specs

I initially changed Articles to require publish_at, but when I thought about it more I realized that wasn't the best option.  Instead, I updated the search in the blogs app because it wasn't filtering non-published articles.

### CODE REVIEW

File: lib/buweb/article.rb [hunk 1]
```diff
@@ -135,28 +137,27 @@ def dont_index?
...
-    end
+    return unless !imported && site_id.blank?
+    errors.add :base, 'Site is required.'
   end
 
   def site_categories_type
-    if site_categories.present? && site_categories.ne(type: "Article").present?
-      errors.add :base, 'All categories must be for articles.'
-    end
+    return unless site_categories.present? &&
```
Comment (rift137) [inline lib/buweb/article.rb]: I wonder, can this be simplified to `return unless site_categories&.ne(type: 'Article').present?`

File: lib/buweb/relationship.rb [hunk 2]
```diff
@@ -22,16 +22,23 @@ class Relationship
...
   field :related_id, type: String
 
+  validates :related_type, :base_type, inclusion: { in: CLEAN_TYPES }
+
   # TODO: When/if adding sort order, you really need a seperate sort index
-  #       for each side of the relationship. So a person could sort the departments
-  #       they are related to and a department can sort the people it is realated to.
+  #       for each side of the relationship. So a person could sort the
+  #       departments
+  #       they are related to and a department can sort the people it is
```
Comment (rift137) [inline lib/buweb/relationship.rb]: collapse this so that it reads better.

Comment (matthewl20) [reply lib/buweb/relationship.rb]: removed

Comment (rift137) [reply lib/buweb/relationship.rb]: still there?

**Pull request merged**